repair/task_manager_module.hh

```diff
@@ -100,6 +100,33 @@ protected:
     virtual std::optional<double> expected_children_number() const override;
 };
 
+class tablet_repair_task_impl : public repair_task_impl {
+private:
+    std::vector<table_id> _table_ids;
+    dht::token_range_vector _ranges;
+    std::unordered_map<dht::token_range, repair_neighbors> _neighbors;
+    locator::effective_replication_map_ptr _erm;
+    optimized_optional<abort_source::subscription> _abort_subscription;
+public:
+    tablet_repair_task_impl(tasks::task_manager::module_ptr module, repair_uniq_id id, std::string keyspace, dht::token_range_vector ranges, std::unordered_map<dht::token_range, repair_neighbors> neighbors, streaming::stream_reason reason, table_id tid, locator::effective_replication_map_ptr erm)
+        : repair_task_impl(module, id.uuid(), id.id, "keyspace", std::move(keyspace), "", "", tasks::task_id::create_null_id(), reason)
```

Comment:
I have no idea what the scope and entity do. I followed existing usage. The table is set to empty in user_requested_repair_task_impl too. I think it is probably not really used in task manager.

Reply:
@Deexie Do you have some hints on this?

Reply:
`scope` indicates the size of the job represented by the task. For `user_requested_repair_task_impl`  its "keyspace" as it covers keyspace repair, here I think "tablet" is ok.

`table` is empty when task represents work on many tables or is unrelated to tables. `tablet_repair_task_impl` works on one table, so it would be nice to pass its name.

`entity` is for additional info and is empty for most tasks.

Reply:
The tablet_repair_task_impl also covers keyspace repair. In my view version, it can repair multiple tables as well. So, I will keep it the same as user_requested_repair_task_impl. Thanks @Deexie .

Reply:
> The tablet_repair_task_impl also covers keyspace repair.

But the task works on a single tablet, so its scope is "tablet", even though its part of a larger keyspace repair. If we tracked keyspace repair as a parent task, then the scope of that task would be "keyspace".

If you set the scope to "keyspace", and repair tablets concurrently, you will see many tasks for the same keyspace with scope "keyspace", which will be confusing, we're not really repairing the same keyspace concurrently.

Reply:
In v2, tablet_repair_task_impl works on multiple tablets, so it is "keyspace" scope. But anyway, I think the concept of scope in task manager is confusing in the first place.

Reply:
Ok.